src/xdist/remote.py

```diff
@@ -56,6 +59,45 @@ def worker_title(title):
         pass
 
 
+class RemoteMessageHandler(logging.Handler):
+    """
+    This handler sends events to a queue. Typically, it would be used together
+    with a multiprocessing Queue to centralise logging to file in one process
+    (in a multi-process application), so as to avoid file write contention
+    between processes.
+
+    This code is new in Python 3.2, but this class can be copy pasted into
+    user code for use with earlier Python versions.
+    """
+
+    def __init__(self, queue):
+        """
+        Initialise an instance, using the passed queue.
+        """
+        logging.Handler.__init__(self)
+        self.queue = queue
+
+    def emit(self, record):
+        """
+        Emit a record.
+
+        Writes the LogRecord to the queue, preparing it for pickling first.
+        """
+        try:
+            msg = self.format(record)
+            # bpo-35726: make copy of record to avoid affecting other handlers in the chain.
+            record = copy.copy(record)
+            record.message = msg
+            record.msg = msg
+            record.args = None
+            record.exc_info = None
+            record.exc_text = None
+            x = pickle.dumps(record)
```

Comment:
Probably does not matter much, but we might force protocol version 4 here to ensure compatibility between different python interpreters.

Reply:
`            x = pickle.dumps(record, protocol=4)`
or
`            x = pickle.dumps(record, protocol=pickle.DEFAULT_PROTOCOL)`
?

Reply:
are log messages pickle safe?
im "pretty" sure there some hazard when dealing with custom objects